pkg/controller/certificatesigningrequest/approver_controller.go

```diff
@@ -0,0 +1,335 @@
+	queue             workqueue.RateLimitingInterface
+	approvers         []approver
+}
+
+// NewCSRApproverController returns a new *CSRApproverController.
+func NewCSRApproverController(client clientset.Interface, csrInformer csrinformer.CertificateSigningRequestInformer, approvePolicy config.AutoApprovePolicy) *CSRApproverController {
+	c := &CSRApproverController{
+		client:            client,
+		csrLister:         csrInformer.Lister(),
+		csrListerSynced:   csrInformer.Informer().HasSynced,
+		autoApprovePolicy: approvePolicy,
+		queue:             workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "certificateSigningRequest"),
+		approvers: []approver{
+			&ipsecCertificateApprover{
+				client:        client,
+				approvePolicy: approvePolicy,
+			},
+		},
+	}
+	csrInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: c.enqueueCertificateSigningRequest,
+			UpdateFunc: func(old, cur interface{}) {
+				c.enqueueCertificateSigningRequest(cur)
+			},
+			DeleteFunc: c.enqueueCertificateSigningRequest,
+		},
+		resyncPeriod,
+	)
+	return c
+}
+
+// Run begins watching and syncing of the CSRApproverController.
+func (c *CSRApproverController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting %s", approverControllerName)
+	defer klog.Infof("Shutting down %s", approverControllerName)
+
+	cacheSyncs := []cache.InformerSynced{c.csrListerSynced}
+	if !cache.WaitForNamedCacheSync(approverControllerName, stopCh, cacheSyncs...) {
+		return
+	}
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (c *CSRApproverController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *CSRApproverController) enqueueCertificateSigningRequest(obj interface{}) {
+	csr, ok := obj.(*certificatesv1.CertificateSigningRequest)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("Received unexpected object: %v", obj)
+			return
+		}
+		csr, ok = deletedState.Obj.(*certificatesv1.CertificateSigningRequest)
+		if !ok {
+			klog.Errorf("DeletedFinalStateUnknown contains non-CertificateSigningRequest object: %v", deletedState.Obj)
+			return
+		}
+	}
+	c.queue.Add(csr.Name)
+}
+
+func (c *CSRApproverController) syncCSR(key string) error {
+	startTime := time.Now()
+	defer func() {
+		d := time.Since(startTime)
+		klog.V(2).Infof("Finished syncing CertificateSigningRequest %s. (%v)", key, d)
+	}()
+
+	csr, err := c.csrLister.Get(key)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if approved, denied := getCertApprovalCondition(&csr.Status); approved || denied {
+		return nil
+	}
+	for _, a := range c.approvers {
+		if a.recognize(csr) {
+			return a.verify(csr)
+		}
+	}
+	klog.Warningf("unrecognized CertificateSigningRequest %s", key)
```

Comment:
If this is not meant for any Antrea signer, it shouldn't even be logged, otherwise it would lead to many unrelated logs?

Reply:
increased logging level.